### PR TITLE
fix(ref-imp): #1169 - fixed bug on commit-reveal chain leading to infinite loop in resolution

### PR DESCRIPTION
**Pull request by thehenrytsai**
Fix to issue raised: https://github.com/decentralized-identity/sidetree/issues/1169

### CODE REVIEW

**Review by xinaxu — DISMISSED** [review]

**Review by OR13 — APPROVED** [review]

File: lib/core/Resolver.ts [hunk 1]
```diff
@@ -104,28 +104,34 @@ export default class Resolver {
    */
   private async applyRecoverAndDeactivateOperations (startingDidState: DidState, commitValueToOperationMap: Map<string, AnchoredOperationModel[]>)
     : Promise<DidState> {
-    let didState = startingDidState;
+    // This stores all commitment hash of the corresponding reveal value of operations that have been successfully applied,
```
Comment (tplooker) [inline lib/core/Resolver.ts:107]:
```suggestion
    // This stores all commitment hashes of the corresponding reveal values of the operations that have been successfully applied,
```

File: lib/core/Resolver.ts [hunk 2]
```diff
@@ -136,22 +142,27 @@ export default class Resolver {
    */
   private async applyUpdateOperations (startingDidState: DidState, commitValueToOperationMap: Map<string, AnchoredOperationModel[]>)
     : Promise<DidState> {
-    let didState = startingDidState;
+    // This stores all commitment hash of the corresponding reveal value of operations that have been successfully applied,
```
Comment (tplooker) [inline lib/core/Resolver.ts:145]:
```suggestion
    // This stores all commitment hashes of the corresponding reveal values of the operations that have been successfully applied,
```

**Review by tplooker — APPROVED** [review]

File: lib/core/Resolver.ts (hunk 1 above)
Comment (thehenrytsai) [reply lib/core/Resolver.ts]: Fixed.

File: lib/core/Resolver.ts (hunk 2 above)
Comment (thehenrytsai) [reply lib/core/Resolver.ts]: Fixed.

**Review by xinaxu — APPROVED** [review]

**Pull request merged**